### DISCUSSION TITLE
Allow logging into AWS ECR using more secure method

### DISCUSSION THREAD

**Discussion by thomaschaaf** [renovatebot/renovate#26498] [Suggest an Idea]
### Tell us more.

In our organization we sadly do not allow creating users with access tokens as they can be leaked and then used in unsecure ways. Github allows the use of [OpenID Connect](https://docs.github.com/en/actions/deployment/security-hardening-your-deployments/configuring-openid-connect-in-amazon-web-services). It would be great if renovatebot would implement a similar method of authentication.

Using openid it should be possible to provide a role to assume.

**Comment by rarkins** [comment]
How are you running Renovate?

**Reply by thomaschaaf** [reply]
I'm using the hosted SaaS version

**Reply by rarkins** [reply]
I'm not sure how that would work (I think OIDC is bidirectional so AWS would need to be provisioned to know about your Renovate instance and maybe even be able to reach it directly). But I'm interested in the approach

**Reply by nitrocode** [reply]
Yes this would be fantastic so we can avoid using an iam access key and secret key

- mend would need to be oidc compatible
- user should create an oidc aws provider for mend.io (idp)
- user's aws then retrieves all of the information from the idp such as the jwks keys which verifies the signed tokens
- user's aws should have an iam role that allows federated role assumption using the oidc provider
- mend should now be able to temporarily assume the iam role without static creds

This page has more info https://docs.aws.amazon.com/IAM/latest/UserGuide/id_roles_providers_create_oidc.html

There is also an oidc-provider library in node

https://www.npmjs.com/package/oidc-provider

**Reply by nitrocode** [reply]
This is a related discussion on oidc

https://github.com/renovatebot/renovate/discussions/23670
